text/3488-deprecate-cargo-categories.md

```diff
@@ -0,0 +1,374 @@
+  - [Discussion on the initial list of categories is started](https://github.com/rust-lang/crates.io/pull/488)
+    - The initial list was adopted from https://github.com/rust-unofficial/awesome-rust
+    - A couple of comments suggest using a list of "official" keywords instead
+      of introducing a parallel concept of categories (https://github.com/rust-lang/crates.io/pull/488#issuecomment-265593195
+      and https://github.com/rust-lang/crates.io/pull/488#issuecomment-265623340)
+
+
+# Statistics
+
+The following numbers have been extracted from a recent
+[crates.io database dump](https://crates.io/data-access#database-dumps):
+
+- 40,649 crates have `categories`
+  - 3,464 of those crates have only `categories`, and no `keywords`
+  - 3,622 of those crates have only a single category
+- 61,216 crates have `keywords`
+  - 24,031 of those crates have only `keywords`, and no `categories`
+  - 4,694 of those crates have only a single keyword
+- 37,185 crates have both `categories` and `keywords`
+- 56,694 crates have neither `categories`, nor `keywords`
+
+
+# Keywords vs. Categories
+
+**Keywords** (aka. tags) describe an open set of names to categorize packages.
+The set being "open" means that users can use arbitrary keywords without having
+to choose from a fixed list of available names.
+
+**Categories** describe a closed set of names to categorize packages. The set 
+being "closed" means that users have to choose from a fixed list of available
+categories, and can't easily add new ones.
+
+**Comparison:**
+
+- **Creation:** As described above, the keywords concept makes it
+quite easy to create new names by simply using them in the `Cargo.toml` file.
+With categories, a pull request to the crates.io repository has to be opened,
+followed by a discussion among the crates.io team on whether there might be
+enough justification for adding such a new category.
+
+- **Typos:** The chance of misspelling categories is reduced since cargo and
+crates.io can warn about misspelled category names. Keywords can be arbitrary,
+so a typo-detection is not quite as easy.
+
+  - It would be possible to compare keywords against a list of commonly used
+  keywords though and warn the user about potential typos.
+
+- **Duplicates:** Since the list of categories is fixed and new additions are
+checked by the crates.io team, the chance for having categories with similar
+purposes is small. With keywords, it is easy to have both `math` and
+`mathematics` keywords available on crates.io. In that case it would be the
+burden of the crate authors to use the most appropriate and/or commonly used
+keywords.
+
+  - This could be countered to some degree by having a list of common aliases
+  on the server-side that are applied either at publish-time or when a user
+  searches for a particular keyword. [Experience from libs.rs indicates that
+  there is no trivial solution to this though](https://github.com/rust-lang/rfcs/pull/3488#issuecomment-1721208225) 
```

Comment:
My take on wording from earlier comment:

```suggestion
- **Duplicates:** Since the list of categories is fixed and new additions are
checked by the crates.io team, the chance for having categories with similar
purposes is small. With keywords, it is easy to have both `math` and
`mathematics` keywords available on crates.io. In that case it would be the
burden of the crate authors to use the most appropriate and/or commonly used
keywords.

  - This could be countered to some degree by having a list of common aliases
  on the server-side that are applied either at publish-time or when a user
  searches for a particular keyword. [Experience from libs.rs indicates that
  there is no trivial solution to this though](https://github.com/rust-lang/rfcs/pull/3488#issuecomment-1721208225) 

  - The concept of duplicates itself is also more ambiguous than you'd think.
  One good example is the endless decision on whether keywords should be
  combined or split up: for example, using "http" and "server" or "http-server"
  keywords. This problem isn't unique to keywords or categories, since there's
  always overlap between different groups and it's never clear where to draw the
  boundaries.
```